internal/tls.go

```diff
@@ -138,7 +138,7 @@ func NewServerCert() (tls.Certificate, []byte, error) {
 		IPAddresses: []net.IP{net.IPv6loopback, net.IPv4(127, 0, 0, 1)},
 		DNSNames:    []string{"localhost"},
 	}
-	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, priv.PublicKey, priv)
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &priv.PublicKey, priv)
```

Comment:
Another subtle mistake: we need to pass a the _address_ of `priv.PublicKey`, because it accepts a value of type `*rsa.PublicKey` but was instead getting a value of type `rsa.PublicKey`.